### ISSUE TITLE
fill does not work as documented

### ISSUE THREAD

**Issue by chris-33** [geraintluff/uri-templates#15]
The following statement

    new UriTemplate('/categories/{id}')
                .fill({ id: 1 });

produces "TypeError: '[object Object]' is not a function (evaluating 'valueFunction(varSpec.name)')", although from my understanding of the [docs](https://github.com/geraintluff/uri-templates#substitution-using-an-object) this is a correct invocation.

**Comment by geraintluff** [comment]
Yeah, that should work.

What version are you using?

**Comment by chris-33** [comment]
Sorry, can't tell anymore...I came across this trying out different candidates for URI templating and ended up using a different one. I was using whatever was the current version when I posted this issue (August 13th).

**Comment by geraintluff** [comment]
Fair enough - worrying, though, as I can't reproduce this.  Were you using NPM or Bower, or what?

**Comment by geraintluff** [comment]
It's possible I'm not publishing to Bower properly, so it might have a super-old version.

If that's the case, this is an extremely useful bug report!

**Comment by brettz9** [comment]
I know this is an ancient issue, but FWIW, with Bower deprecated, I think we might want to actually just remove Bower completely by now.
